cli/command/stack/kubernetes/conversion.go

```diff
@@ -122,48 +156,71 @@ const (
 )
 
 // Replicas conversion
-func replicasToServices(replicas *appsv1beta2.ReplicaSetList, services *apiv1.ServiceList) ([]swarm.Service, map[string]formatter.ServiceListInfo, error) {
-	result := make([]swarm.Service, len(replicas.Items))
-	infos := make(map[string]formatter.ServiceListInfo, len(replicas.Items))
+func replicasToServices(replicas *appsv1beta2.ReplicaSetList, daemons *appsv1beta2.DaemonSetList, services *apiv1.ServiceList) ([]swarm.Service, map[string]formatter.ServiceListInfo, error) {
+	l := len(replicas.Items) + len(daemons.Items)
+	result := make([]swarm.Service, l)
+	infos := make(map[string]formatter.ServiceListInfo, l)
 	for i, r := range replicas.Items {
-		serviceName := r.Labels[labels.ForServiceName]
-		serviceHeadless, ok := findService(services, serviceName)
-		if !ok {
-			return nil, nil, fmt.Errorf("could not find service '%s'", serviceName)
-		}
-		stack, ok := serviceHeadless.Labels[labels.ForStackName]
-		if ok {
-			stack += "_"
-		}
-		uid := string(serviceHeadless.UID)
-		s := swarm.Service{
-			ID: uid,
-			Spec: swarm.ServiceSpec{
-				Annotations: swarm.Annotations{
-					Name: stack + serviceHeadless.Name,
-				},
-				TaskTemplate: swarm.TaskSpec{
-					ContainerSpec: &swarm.ContainerSpec{
-						Image: getContainerImage(r.Spec.Template.Spec.Containers),
-					},
-				},
-			},
-		}
-		if serviceNodePort, ok := findService(services, serviceName+publishedOnRandomPortSuffix); ok && serviceNodePort.Spec.Type == apiv1.ServiceTypeNodePort {
-			s.Endpoint = serviceEndpoint(serviceNodePort, swarm.PortConfigPublishModeHost)
-		}
-		if serviceLoadBalancer, ok := findService(services, serviceName+publishedServiceSuffix); ok && serviceLoadBalancer.Spec.Type == apiv1.ServiceTypeLoadBalancer {
-			s.Endpoint = serviceEndpoint(serviceLoadBalancer, swarm.PortConfigPublishModeIngress)
+		s, uid, err := replicaToService(r.Labels, services, r.Spec.Template.Spec.Containers)
+		if err != nil {
+			return nil, nil, err
 		}
-		result[i] = s
+		result[i] = *s
 		infos[uid] = formatter.ServiceListInfo{
 			Mode:     "replicated",
 			Replicas: fmt.Sprintf("%d/%d", r.Status.AvailableReplicas, r.Status.Replicas),
 		}
 	}
+	for i, d := range daemons.Items {
+		s, uid, err := replicaToService(d.Labels, services, d.Spec.Template.Spec.Containers)
+		if err != nil {
+			return nil, nil, err
+		}
+		result[i+len(replicas.Items)] = *s
+		infos[uid] = formatter.ServiceListInfo{
+			Mode:     "global",
+			Replicas: fmt.Sprintf("%d/%d", d.Status.NumberReady, d.Status.DesiredNumberScheduled),
+		}
+	}
+	sort.Slice(result, func(i, j int) bool {
+		return result[i].ID < result[j].ID
+	})
 	return result, infos, nil
 }
 
+func replicaToService(labls map[string]string, services *apiv1.ServiceList, containers []apiv1.Container) (*swarm.Service, string, error) {
```

Comment:
~~Wondering: Wwy pass labels and containers separately? Both are properties of the replica~~

Ahm, right, so this function is used for converting both `Daemonsets` and `Replicasets`, which are very similar, but different types

- Nit: rename this function to (e.g.) `convertToService()` (as it doesn't only convert replicas)
- We use `labls` only for the service name; may be better to just pass the name?
- Why return both the `ID` and the Service here? (i.e. can use `Service.ID` to get the `uid`)

Service signature could look like;

```go
func convertToService(serviceName string, services *apiv1.ServiceList, containers []apiv1.Container) (*swarm.Service, error) {
```

Then in the loop, this could do;

```go
infos[s.ID] = formatter.ServiceListInfo{
	Mode:     "replicated",
	Replicas: fmt.Sprintf("%d/%d", r.Status.AvailableReplicas, r.Status.Replicas),
}
```